File: python/1-10/Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @return a ListNode
    def addTwoNumbers(self, l1, l2):
        r=0
        array=[]
        while l1!=None or l2!=None:
            if l1==None:
                num1=0 
            else :
                num1=l1.val
            if l2==None:
                num2=0 
            else:
                num2=l2.val
            sum = num1+num2+r
            r = sum/10
            data = sum%10
            array.append(data)
            if l1!=None:
                l1=l1.next
            if l2!=None:
                l2=l2.next
        if r!=0:
            array.append(1)
        head=ListNode(0)
        p=head
        for data in array:
            temp=ListNode(data)
            p.next=temp
            p=temp
        return head.next

File: Cpp/1-10/Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        if(!l1) return l2;
        if(!l2) return l1;
        int carry=0,sum,a,b;
        ListNode thead(0);
        ListNode* temp= &thead;
        while(l2!=NULL ||l1!=NULL){
            a=(l1==NULL)?0:l1.val;
            b=(l2==NULL)?0:l2.val;
            sum = a+b+carry;
            carry = sum/10;
            sum = sum%10;
            ListNode* t =new ListNode(sum);
            temp =temp->next = t;
            if(l1) l1=l1->next;
            if(l2) l2 = l2->next;
            
        }
        if(carry!=0){
            ListNode* t =new ListNode(carry); 
            temp->next = t;
        }
        return thead.next;
    }
};
